engine/dlib/src/dlib/hash.cpp

```diff
@@ -665,12 +666,24 @@ DM_DLLEXPORT void dmHashReverseErase64(uint64_t hash)
 DM_DLLEXPORT const char* dmHashReverseSafe64(uint64_t hash)
 {
     const char* s = (const char*)dmHashReverse64(hash, 0);
-    return s != 0 ? s : "<unknown>";
+    if (s == 0)
+    {
+        char tmp[64];
+        dmSnPrintf(tmp, sizeof(tmp), "<unknown:%llu>", (unsigned long long)hash);
```

Comment:
What is the output used for?
Debugging only?
Or as actual strings in a release build? (Do we want to encourage that?)

And, what's the best format here?
signed long long, unsigned long long, hex?

Are we using the hashes elsewhere? E.g. in an editor/bob script, what would be the output format?

Reply:
you right

What I want is to have an ability to get a value from a hashmap by this value in debug build
I expect hash to be the same in release and in debug build
So if I have the hash value from the release build, it should be possible to get the corresponding string value in the debug build
---
Talking about downsides. Yes, people already try to extract strings from the hash value, and this fix will not make it worse. Also, a string from debug will be different from a string from release, so they can’t abuse it

Reply:
One more thing is: 
I have a lot of messages in a released game like:
>“DEFOLD:ERROR:GAMEOBJECT: Instance ‘<unknown>’ could not be found when dispatching message ‘<unknown>’ sent from game:<unknown>#<unknown>\n”

It makes it impossible to figure out what’s going on. I hope having such a numeric value will help with "symbolication" of this message to make it easier to understand what's going on there